src/OpenTracing.BasicTracer/Tracer.cs

```diff
@@ -46,29 +46,55 @@ public ISpan StartSpan(string operationName, StartSpanOptions options)
             return span;
         }
 
-        public void Inject(ISpanContext spanContext, IInjectCarrier carrier)
+        public void Inject(ISpanContext spanContext, string format, IInjectCarrier carrier)
```

Comment:
been a few years since I've done C#, but can't the `format` argument be a typed constant? E.g.

```
    public interface Format<T> { }
    
    public class StringFormat<T>: Format<T> {
        private string code;
        public StringFormat(string code) {
            this.code = code;
        }
    }
    
    public class Builtin {
        // using `int` as carrier type here
        public static StringFormat<int> TextMap = new StringFormat<int>("buildin.textmap");
    }
```